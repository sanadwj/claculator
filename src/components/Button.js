import React from 'react';
import PropTypes from 'prop-types';
import './App.css';

import styled from 'styled-components';

const Button = props => {
  const handleClick = buttonName => {
    props.handleClick(buttonName);
  };

  const { name } = props;
  const ButtonText = styled.button`
  width: ${props => (props.name === '0' ? '50%' : '25%')};
  height: 100px;
  background-color: ${props => (props.name === '-' || props.name === 'X' || props.name === '+' || props.name === '=' || props.name === 'รท' ? 'orange' : 'lightgray')};
`;

  return (
    <ButtonText name={name} className="button" onClick={() => { handleClick(name); }} type="button">
      <h3>{name}</h3>
    </ButtonText>
  );
};

Button.propTypes = {
  name: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default Button;
