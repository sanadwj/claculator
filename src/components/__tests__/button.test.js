import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import Button from '../Button';

describe('Button', () => {
  const handleClick = () => ({ total: '', next: '', operation: '' });
  it('renders correctly', () => {
    const tree = renderer.create(
      <Button name="0" onClick={handleClick} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Creates a component', () => {
    render(
      <Button name="0" onClick={handleClick} />,
    );
    const element = screen.getByText('0');
    expect(element).toBeInTheDocument();
  });
});
