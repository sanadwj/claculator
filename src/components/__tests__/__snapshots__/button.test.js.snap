// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="sc-bdfBwQ AFkPE button"
  name="0"
  onClick={[Function]}
  type="button"
>
  <h3>
    0
  </h3>
</button>
`;
